src/Chainweb/Pact/PactService.hs

```diff
@@ -284,20 +288,67 @@ serviceRequests memPoolAccess reqQ = do
 
     toPactInternalError e = Left $ PactInternalError $ T.pack $ show e
 
-    tryOne which mvar m = tryOne' which mvar (return . Right) m
-
-    tryOne' which mvar post m = do
-        r <- try m
-        case r of
-            Left (SomeException e) -> do
-                logError $ mconcat [ "Received exception running pact service ("
-                                   , which
-                                   , "): "
-                                   , show e ]
-                liftIO $ putMVar mvar $! toPactInternalError e
-            Right r' -> post r' >>= liftIO . putMVar mvar
-
-
+    tryOne
+        :: String
+        -> MVar (Either PactException a)
+        -> PactServiceM cas a
+        -> PactServiceM cas ()
+    tryOne which mvar m = tryOne' which mvar Right m
+
+    tryOne'
+        :: String
+        -> MVar (Either PactException b)
+        -> (a -> Either PactException b)
+        -> PactServiceM cas a
+        -> PactServiceM cas ()
+    tryOne' which mvar post m =
+        (evalPact (post <$> m) >>= (liftIO . putMVar mvar))
```

Comment:
`evalPact` the name clashes with some Pact constructs that we use. Can we use a different name for this (e.g. `eval_`)?